src/book/02-system/01-foundations/07-director.mdx

```diff
@@ -0,0 +1,442 @@
+
+```cpp
+// Emit a Task requesting to walk forwards
+emit<Task>(std::make_unique<Walk>(Eigen::Vector3d(1.0, 0.0, 0.0)));
+```
+
+This walk engine Task contains information on the speed and direction that the robot should walk in. Other Tasks, such as `GetUpWhenFallen` does not need any data.
+
+Tasks can take four arguments, as described in the following table in order.
+
+| Argument | Description                                                                                                                                                                                                                                                                                                                                                              |
+| -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| data     | A pointer to a Protobuf message, describing the Task.                                                                                                                                                                                                                                                                                                                    |
+| priority | A positive integer. Higher numbers indicate higher priority. If two Tasks have the same priority, then the one emitted first will be chosen. If no priority is specified, then the default priority of 0 is used.                                                                                                                                                        |
+| optional | A boolean. If true, then the Task is considered optional. If two Tasks need the same resources and have the same priority but one is optional, then the non-optional Task will be chosen. Optional Tasks do not need to be active for other Tasks from the same Provider to run. If no optional argument is specified, then the default optional value of false is used. |
+| name     | A string. If provided, then this string will be used when logging messages. The default is the empty string.                                                                                                                                                                                                                                                             |
+
+### Root Level Tasks
+
+When a Task is emitted from a reaction which is not itself a Provider, that Task is considered a root level task.
+These root level tasks are all considered siblings within the priority comparison system at the root level of the graph.
+When a root level task is emitted with a `nullptr` is removed from the list of Tasks.
```

Comment:
```suggestion
When a root level task is emitted with a `nullptr`, that root level task is removed from the Director tree and the list of Tasks to execute.
```
Not sure if this is better. The idea is that you can remove tasks in providers (ie non root level tasks) by just not emitting them when you next run the provider. But for root-level, there is no provider to do this from, since they originate outside any provider. The way to remove them from the director system is to emit the task again with a nullptr.

Reply:
Yeah that's definitely clearer.